### PR TITLE
Always serve the settings profile from /profile

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes https://github.com/vector-im/riot-web/issues/9551

This will obviously incur a performance cost on the server (albeit hopefully minimal) and will make the dialog slower. However, it will ensure that it is accurate (see bug for why this isn't accurate).

### CODE REVIEW

**Comment by t3chguy** [comment]
I would argue the fix should be ensuring that js-sdk's User objects are not mangled improperly by `m.room.member` events rather than relying on non-syncing things :((

**Comment by Half-Shot** [comment]
I think our ability to distinguish global and non global member profiles across federation w/o presence is pretty much fudged at this point. We need to fix /sync so that we can actually determine this for ourselves, or we need to do this.

File: src/components/views/settings/ProfileSettings.js [hunk 1]
```diff
@@ -27,27 +27,32 @@ export default class ProfileSettings extends React.Component {
...
         this.state = {
             userId: user.userId,
-            originalDisplayName: user.displayName,
-            displayName: user.displayName,
-            originalAvatarUrl: avatarUrl,
-            avatarUrl: avatarUrl,
+            originalDisplayName: undefined,
+            displayName: undefined,
+            originalAvatarUrl: undefined,
+            avatarUrl: undefined,
```
Comment (turt2live) [inline src/components/views/settings/ProfileSettings.js]: we typically use `null` instead

**Comment by turt2live** [comment]
I would also argue that we should fix the js-sdk ftr - this is code that affects everyone.

**Comment by Half-Shot** [comment]
>I would also argue that we should fix the js-sdk ftr

Sure, if it was just as simple as just fixing the SDK that then I'd agree. For the reason given above, we can't fix it in the SDK alone. The JS-SDK is using membership events to cache profiles because there is no other way to determine global profile changes in Matrix, without using presence, which we can't rely on.

Fixing this bug would mean using an alternative technique for tracking user's global profiles:
- Stop caching, and rely on `/profile` entirely. **v.expensive**.
- Cache, but lookup `/profile` on each membership change to check if the profile change is global. **expensive**
- Cache once per session, and do not attempt to track. **cheap-ish, but rubbish**
- Use the room membership, effectively not supporting per room profiles. **cheap, but flawed**. This is what we do now.
- Fix presence, and use that. **Unlikely**
- Fix presence enough to make it emit when the profile is changed. **probably ideal, but synapse change required**
- Use an alternative (EDUs perhaps?) means to report profile changes. **spec land**

File: src/components/views/settings/ProfileSettings.js (hunk 1 above)
Comment (Half-Shot) [reply src/components/views/settings/ProfileSettings.js]: [Except when not?](https://github.com/matrix-org/matrix-react-sdk/pull/3739/files/370b7d45119b35bfc087582338d67946c89c2dc4#diff-d23966825df49bff164e0186b877ad39R67)

**Comment by turt2live** [comment]
There's also the argument that per-room profiles aren't a real thing (the spec has a number of open bug that are blocking its addition to the spec: https://github.com/matrix-org/matrix-doc/issues/1376 and refs).

Given the options above, spec land feels like the least worst option. I don't think it's fair for us to fix profiles at the app layer when there's a large number of js-sdk consumers which might rely on this working (and historically when we don't use a feature of the js-sdk, it dies).

**Comment by Half-Shot** [comment]
>Fix presence enough to make it emit when the profile is changed. probably ideal, but synapse change required

I remember now that this is already a thing as per the spec, but obviously matrix.org doesn't support presence (and neither do some of the larger modular customers). I think we are unlikely to add a new EDU to the spec which replicates a feature we disabled on larger synapses. This is a pit of sadness indeed :(

**Comment by t3chguy** [comment]
This has been made redundant by the OwnProfileStore which polls `/profile` whenever it thinks it has become stale

**Pull request closed without merging**